src/analyze.cpp

```diff
@@ -5149,6 +5149,12 @@ static bool const_values_equal_array(CodeGen *g, ConstExprValue *a, ConstExprVal
 }
 
 bool const_values_equal(CodeGen *g, ConstExprValue *a, ConstExprValue *b) {
+    if (a == nullptr || b == nullptr) {
```

Comment:
this function previously asserted that `a != nullptr && b != nullptr` (via the `assert(a->type->id == b->type->id);` assertion). I would like to keep it that way. It makes debugging a lot easier.

Reply:
OK, yeah it can be done that way too.